lib/OpenQA/Worker/Cache.pm

```diff
@@ -41,82 +47,55 @@ sub new {
     shift->SUPER::new(@_)->init;
 }
 
+sub from_worker {
+    my ($worker_settings, undef) = OpenQA::Worker::Common::read_worker_config(undef, undef);
+    __PACKAGE__->new(
+        host     => 'localhost',
+        location => ($ENV{CACHE_DIR} || $worker_settings->{CACHEDIRECTORY}),
+        (limit => int($worker_settings->{CACHELIMIT}) * (1024**3)) x !!($worker_settings->{CACHELIMIT}));
+}
+
 sub DESTROY {
     my $self = shift;
 
-    $self->dbh->disconnect() if $self->dbh;
+    $self->dbh->db->disconnect() if $self->dbh;
 }
 
 sub deploy_cache {
     my $self = shift;
     local $/;
     my $sql = <DATA>;
-    print STDOUT "\n\n\n\nINIT\n";
     log_info "Creating cache directory tree for " . $self->location;
     remove_tree($self->location, {keep_root => 1});
-    make_path(File::Spec->catdir($self->location, Mojo::URL->new($self->host)->host || $self->host));
-    make_path(File::Spec->catdir($self->location, 'tmp'));
+    path($self->location)->make_path;
+    path($self->location, 'tmp')->make_path;
 
-    log_info "Deploying DB: $sql";
+    log_info "Deploying DB: $sql (dsn " . $self->dsn . ")";
 
-    my $dbh = DBI->connect($self->dsn, undef, undef, {RaiseError => 1, PrintError => 1, AutoCommit => 0})
-      or die("Could not connect to the dbfile.");
-    $dbh->do($sql);
-    $dbh->commit;
-    $dbh->disconnect;
-    $self->dbh($dbh);
+    $self->dbh(Mojo::SQLite->new($self->dsn)) or die("Could not connect to the dbfile.");
+    my $tx = $self->dbh->db->begin;
+    $self->dbh->db->query($sql);
+    $tx->commit;
```

Comment:
Another case where you need to hold on to the db handle instead of calling `$self->dbh->db` repeatedly.